lib/src/components/component.dart

```diff
@@ -87,5 +87,5 @@ abstract class Component {
   ///   myImage = await gameRef.load('my_image.png');
   /// }
   /// ```
-  Future<void> onLoad() => null;
+  Future<void>? onLoad() => null;
```

Comment:
Does this makes sense? How can a void be optional?

I haven't tested, But I think we could instead do this:

```dart
Future<void> onLoad() => Future.value();
```

Reply:
we discussed in the past right, void is the way to guaranteed it's added immediatly

Reply:
Sorry, I think my initial comment was confusing and that made you confuse things, I am wondering about the return value being a null instead of always being a future, but either way I now realize that this has nothing to do with null safety, so don't worry about it.